### PR TITLE
Check IsBaselineSimdIsaSupported in LocalAddressVisitor

### LINKED ISSUE #90083: Test failure JIT/SIMD/VectorConvert_ro_Target_64Bit/VectorConvert_ro_Target_64Bit.cmd
**Failed in:** [runtime-coreclr jitstress-isas-arm 20230805.1](https://dev.azure.com/dnceng-public/public/_build/results?buildId=364705&view=ms.vss-test-web.build-test-results-tab&runId=7710296&resultId=101614&paneView=debug)

**Failed tests:**
```
coreclr linux arm64 Checked jitstress_isas_nohwintrinsic @ (Ubuntu.1804.Arm64.Open)Ubuntu.1804.Armarch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8
    - JIT/SIMD/VectorConvert_ro_Target_64Bit/VectorConvert_ro_Target_64Bit.cmd
    - JIT/SIMD/VectorConvert_r_Target_64Bit/VectorConvert_r_Target_64Bit.cmd
coreclr windows arm64 Checked jitstress_isas_nohwintrinsic @ Windows.11.Arm64.Open
    - JIT/SIMD/VectorConvert_ro_Target_64Bit/VectorConvert_ro_Target_64Bit.cmd
    - JIT/SIMD/VectorConvert_r_Target_64Bit/VectorConvert_r_Target_64Bit.cmd
coreclr osx arm64 Checked jitstress_isas_nohwintrinsic @ OSX.1200.ARM64.Open
    - JIT/SIMD/VectorConvert_ro_Target_64Bit/VectorConvert_ro_Target_64Bit.cmd
    - JIT/SIMD/VectorConvert_r_Target_64Bit/VectorConvert_r_Target_64Bit.cmd
```

**Error message:**
```
 Assert failure(PID 57 [0x00000039], Thread: 57 [0x0039]): Assertion failed 'compiler->compIsaSupportedDebugOnly(HWIntrinsicInfo::lookupIsa(intrin.id))' in 'VectorTest+VectorConvertTest:VectorConvertDoubleInt64(System.Numerics.Vector`1[double]):int' during 'Generate code' (IL size 245; hash 0x2958de27; FullOpts)

    File: /__w/1/s/src/coreclr/jit/hwintrinsiccodegenarm64.cpp Line: 210
    Image: /root/helix/work/correlation/corerun

apply_reg_state: ip and cfa unchanged; stopping here (ip=0xff2707cc07a0)
apply_reg_state: ip and cfa unchanged; stopping here (ip=0xff2707cc07a0)
waitpid() returned successfully (wstatus 00000000) WEXITSTATUS 0 WTERMSIG 0
/root/helix/work/workitem/e/JIT/SIMD/JIT.SIMD/../VectorConvert_ro_Target_64Bit/VectorConvert_ro_Target_64Bit.sh: line 438:    57 Aborted                 (core dumped) $LAUNCHER $ExePath "${CLRTestExecutionArguments[@]}"

Return code:      1
Raw output file:      /root/helix/work/workitem/uploads/VectorConvert_ro_Target_64Bit/output.txt
Raw output:
BEGIN EXECUTION
/root/helix/work/correlation/corerun -p System.Reflection.Metadata.MetadataUpdater.IsSupported=false -p System.Runtime.Serialization.EnableUnsafeBinaryFormatterSerialization=true VectorConvert_ro_Target_64Bit.dll ''
[createdump] Gathering state for process 57 corerun
[createdump] Crashing thread 0039 signal 6 (0006)
[createdump] Writing crash report to file /home/helixbot/dotnetbuild/dumps/coredump.57.dmp.crashreport.json
[createdump] Crash report successfully written
[createdump] Writing minidump with heap to file /home/helixbot/dotnetbuild/dumps/coredump.57.dmp
[createdump] Written 63246336 bytes (15441 pages) to core file
[createdump] Target process is alive
[createdump] Dump successfully written in 213ms
Expected: 100
Actual: 134
END EXECUTION - FAILED
Test failed. Trying to see if dump file was created in /home/helixbot/dotnetbuild/dumps since 8/5/2023 8:41:03 PM
Processing /home/helixbot/dotnetbuild/dumps/coredump.57.dmp.crashreport.json
Printing stacktrace from '/home/helixbot/dotnetbuild/dumps/coredump.57.dmp.crashreport.json'
Invoking llvm-symbolizer --pretty-print
Errors while running llvm-symbolizer --pretty-print
System.ComponentModel.Win32Exception (2): An error occurred trying to start process 'llvm-symbolizer' with working directory '/root/helix/work/workitem/e/JIT/SIMD/JIT.SIMD'. No such file or directory
   at System.Diagnostics.Process.ForkAndExecProcess(ProcessStartInfo startInfo, String resolvedFilename, String[] argv, String[] envp, String cwd, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec) in /_/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs:line 496
   at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo) in /_/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs:line 456
   at CoreclrTestLib.CoreclrTestWrapperLib.TryPrintStackTraceFromCrashReport(String crashReportJsonFile, TextWriter outputWriter)
Test Harness Exitcode is : 1
To run the test:
Set up CORE_ROOT and run.
> /root/helix/work/workitem/e/JIT/SIMD/JIT.SIMD/../VectorConvert_ro_Target_64Bit/VectorConvert_ro_Target_64Bit.sh
Expected: True
Actual:   False
```

**Stack trace:**
```
   at Xunit.Assert.True(Nullable`1 condition, String userMessage) in /_/src/xunit.assert/Asserts/BooleanAsserts.cs:line 132
   at TestLibrary.OutOfProcessTest.RunOutOfProcessTest(String assemblyPath)
   at Program.<<Main>$>g__TestExecutor114|0_115(StreamWriter tempLogSw, StreamWriter statsCsvSw, <>c__DisplayClass0_0&)
```

Queued | Pipeline | Pipeline_Configuration | OS | Arch | Test | Outcome | Properties | BuildReason
-- | -- | -- | -- | -- | -- | -- | -- | --
2023-08-05T20:51:24.382Z | runtime-coreclr jitstress-isas-arm | Checked-jitstress_isas_nohwintrinsic | osx.1200.arm64.open | arm64 | JIT/SIMD/VectorConvert_ro_Target_64Bit/VectorConvert_ro_Target_64Bit.cmd | Failed | {"System.PhaseName":"run_test_p1__osx_arm64_checked","System.JobAttempt":"1","DefinitionId":"116","CollectionUri":"https://dev.azure.com/dnceng-public/","System.StageAttempt":"1","AzurePipelinesTestRunId":"7710534","Project":"public","DefinitionName":"runtime-coreclr jitstress-isas-arm","Reason":"Schedule","operatingSystem":"OSX.1200.ARM64.Open","configuration":"Checked-jitstress_isas_nohwintrinsic","architecture":"arm64","System.JobName":"__default","System.JobId":"765fd5e9-e47f-5cb0-4cbf-499eaaaee93e","System.PhaseAttempt":"1","BuildId":"364705","BuildNumber":"20230805.1","System.StageName":"Build"} | Schedule
2023-08-05T20:51:24.382Z | runtime-coreclr jitstress-isas-arm | Checked-jitstress_isas_nohwintrinsic | osx.1200.arm64.open | arm64 | JIT/SIMD/VectorConvert_ro_Target_64Bit/VectorConvert_ro_Target_64Bit.cmd | Failed | {"System.PhaseName":"run_test_p1__osx_arm64_checked","System.JobAttempt":"1","DefinitionId":"116","CollectionUri":"https://dev.azure.com/dnceng-public/","System.StageAttempt":"1","AzurePipelinesTestRunId":"7710534","Project":"public","DefinitionName":"runtime-coreclr jitstress-isas-arm","Reason":"Schedule","operatingSystem":"OSX.1200.ARM64.Open","configuration":"Checked-jitstress_isas_nohwintrinsic","architecture":"arm64","System.JobName":"__default","System.JobId":"765fd5e9-e47f-5cb0-4cbf-499eaaaee93e","System.PhaseAttempt":"1","BuildId":"364705","BuildNumber":"20230805.1","System.StageName":"Build"} | Schedule
2023-08-05T20:51:24.382Z | runtime-coreclr jitstress-isas-arm | Checked-jitstress_isas_nohwintrinsic | osx.1200.arm64.open | arm64 | JIT/SIMD/VectorConvert_ro_Target_64Bit/VectorConvert_ro_Target_64Bit.cmd | Failed | {"System.PhaseName":"run_test_p1__osx_arm64_checked","System.JobAttempt":"1","DefinitionId":"116","CollectionUri":"https://dev.azure.com/dnceng-public/","System.StageAttempt":"1","AzurePipelinesTestRunId":"7710534","Project":"public","DefinitionName":"runtime-coreclr jitstress-isas-arm","Reason":"Schedule","operatingSystem":"OSX.1200.ARM64.Open","configuration":"Checked-jitstress_isas_nohwintrinsic","architecture":"arm64","System.JobName":"__default","System.JobId":"765fd5e9-e47f-5cb0-4cbf-499eaaaee93e","System.PhaseAttempt":"1","BuildId":"364705","BuildNumber":"20230805.1","System.StageName":"Build"} | Schedule
2023-08-05T20:51:24.382Z | runtime-coreclr jitstress-isas-arm | Checked-jitstress_isas_nohwintrinsic | osx.1200.arm64.open | arm64 | JIT/SIMD/VectorConvert_ro_Target_64Bit/VectorConvert_ro_Target_64Bit.cmd | Failed | {"System.PhaseName":"run_test_p1__osx_arm64_checked","System.JobAttempt":"1","DefinitionId":"116","CollectionUri":"https://dev.azure.com/dnceng-public/","System.StageAttempt":"1","AzurePipelinesTestRunId":"7710534","Project":"public","DefinitionName":"runtime-coreclr jitstress-isas-arm","Reason":"Schedule","operatingSystem":"OSX.1200.ARM64.Open","configuration":"Checked-jitstress_isas_nohwintrinsic","architecture":"arm64","System.JobName":"__default","System.JobId":"765fd5e9-e47f-5cb0-4cbf-499eaaaee93e","System.PhaseAttempt":"1","BuildId":"364705","BuildNumber":"20230805.1","System.StageName":"Build"} | Schedule
2023-08-05T20:51:24.382Z | runtime-coreclr jitstress-isas-arm | Checked-jitstress_isas_nohwintrinsic | osx.1200.arm64.open | arm64 | JIT/SIMD/VectorConvert_ro_Target_64Bit/VectorConvert_ro_Target_64Bit.cmd | Failed | {"System.PhaseName":"run_test_p1__osx_arm64_checked","System.JobAttempt":"1","DefinitionId":"116","CollectionUri":"https://dev.azure.com/dnceng-public/","System.StageAttempt":"1","AzurePipelinesTestRunId":"7710534","Project":"public","DefinitionName":"runtime-coreclr jitstress-isas-arm","Reason":"Schedule","operatingSystem":"OSX.1200.ARM64.Open","configuration":"Checked-jitstress_isas_nohwintrinsic","architecture":"arm64","System.JobName":"__default","System.JobId":"765fd5e9-e47f-5cb0-4cbf-499eaaaee93e","System.PhaseAttempt":"1","BuildId":"364705","BuildNumber":"20230805.1","System.StageName":"Build"} | Schedule
2023-08-05T20:51:24.382Z | runtime-coreclr jitstress-isas-arm | Checked-jitstress_isas_nohwintrinsic | osx.1200.arm64.open | arm64 | JIT/SIMD/VectorConvert_ro_Target_64Bit/VectorConvert_ro_Target_64Bit.cmd | Failed | {"System.PhaseName":"run_test_p1__osx_arm64_checked","System.JobAttempt":"1","DefinitionId":"116","CollectionUri":"https://dev.azure.com/dnceng-public/","System.StageAttempt":"1","AzurePipelinesTestRunId":"7710534","Project":"public","DefinitionName":"runtime-coreclr jitstress-isas-arm","Reason":"Schedule","operatingSystem":"OSX.1200.ARM64.Open","configuration":"Checked-jitstress_isas_nohwintrinsic","architecture":"arm64","System.JobName":"__default","System.JobId":"765fd5e9-e47f-5cb0-4cbf-499eaaaee93e","System.PhaseAttempt":"1","BuildId":"364705","BuildNumber":"20230805.1","System.StageName":"Build"} | Schedule
2023-08-05T20:51:24.382Z | runtime-coreclr jitstress-isas-arm | Checked-jitstress_isas_nohwintrinsic | osx.1200.arm64.open | arm64 | JIT/SIMD/VectorConvert_ro_Target_64Bit/VectorConvert_ro_Target_64Bit.cmd | Failed | {"System.PhaseName":"run_test_p1__osx_arm64_checked","System.JobAttempt":"1","DefinitionId":"116","CollectionUri":"https://dev.azure.com/dnceng-public/","System.StageAttempt":"1","AzurePipelinesTestRunId":"7710534","Project":"public","DefinitionName":"runtime-coreclr jitstress-isas-arm","Reason":"Schedule","operatingSystem":"OSX.1200.ARM64.Open","configuration":"Checked-jitstress_isas_nohwintrinsic","architecture":"arm64","System.JobName":"__default","System.JobId":"765fd5e9-e47f-5cb0-4cbf-499eaaaee93e","System.PhaseAttempt":"1","BuildId":"364705","BuildNumber":"20230805.1","System.StageName":"Build"} | Schedule
2023-08-05T20:51:24.382Z | runtime-coreclr jitstress-isas-arm | Checked-jitstress_isas_nohwintrinsic | osx.1200.arm64.open | arm64 | JIT/SIMD/VectorConvert_ro_Target_64Bit/VectorConvert_ro_Target_64Bit.cmd | Failed | {"System.PhaseName":"run_test_p1__osx_arm64_checked","System.JobAttempt":"1","DefinitionId":"116","CollectionUri":"https://dev.azure.com/dnceng-public/","System.StageAttempt":"1","AzurePipelinesTestRunId":"7710534","Project":"public","DefinitionName":"runtime-coreclr jitstress-isas-arm","Reason":"Schedule","operatingSystem":"OSX.1200.ARM64.Open","configuration":"Checked-jitstress_isas_nohwintrinsic","architecture":"arm64","System.JobName":"__default","System.JobId":"765fd5e9-e47f-5cb0-4cbf-499eaaaee93e","System.PhaseAttempt":"1","BuildId":"364705","BuildNumber":"20230805.1","System.StageName":"Build"} | Schedule
2023-08-05T20:51:24.382Z | runtime-coreclr jitstress-isas-arm | Checked-jitstress_isas_nohwintrinsic | osx.1200.arm64.open | arm64 | JIT/SIMD/VectorConvert_ro_Target_64Bit/VectorConvert_ro_Target_64Bit.cmd | Failed | {"System.PhaseName":"run_test_p1__osx_arm64_checked","System.JobAttempt":"1","DefinitionId":"116","CollectionUri":"https://dev.azure.com/dnceng-public/","System.StageAttempt":"1","AzurePipelinesTestRunId":"7710534","Project":"public","DefinitionName":"runtime-coreclr jitstress-isas-arm","Reason":"Schedule","operatingSystem":"OSX.1200.ARM64.Open","configuration":"Checked-jitstress_isas_nohwintrinsic","architecture":"arm64","System.JobName":"__default","System.JobId":"765fd5e9-e47f-5cb0-4cbf-499eaaaee93e","System.PhaseAttempt":"1","BuildId":"364705","BuildNumber":"20230805.1","System.StageName":"Build"} | Schedule
2023-08-05T20:51:24.382Z | runtime-coreclr jitstress-isas-arm | Checked-jitstress_isas_nohwintrinsic | osx.1200.arm64.open | arm64 | JIT/SIMD/VectorConvert_ro_Target_64Bit/VectorConvert_ro_Target_64Bit.cmd | Failed | {"System.PhaseName":"run_test_p1__osx_arm64_checked","System.JobAttempt":"1","DefinitionId":"116","CollectionUri":"https://dev.azure.com/dnceng-public/","System.StageAttempt":"1","AzurePipelinesTestRunId":"7710534","Project":"public","DefinitionName":"runtime-coreclr jitstress-isas-arm","Reason":"Schedule","operatingSystem":"OSX.1200.ARM64.Open","configuration":"Checked-jitstress_isas_nohwintrinsic","architecture":"arm64","System.JobName":"__default","System.JobId":"765fd5e9-e47f-5cb0-4cbf-499eaaaee93e","System.PhaseAttempt":"1","BuildId":"364705","BuildNumber":"20230805.1","System.StageName":"Build"} | Schedule

### PR DESCRIPTION
**Pull request by EgorBo**
Fixes https://github.com/dotnet/runtime/issues/90083

### CODE REVIEW

**Comment by EgorBo** [comment]
PTAL @jakobbotsch it seems to be a fallout from https://github.com/dotnet/runtime/pull/89520 (other conditions in this file check for `IsBaselineSimdIsaSupported`)

File: src/coreclr/jit/lclmorph.cpp [hunk 1]
```diff
@@ -1188,15 +1188,16 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
...
+                        m_compiler->IsBaselineSimdIsaSupported())
                     {
                         return isDef ? IndirTransform::WithElement : IndirTransform::GetElement;
                     }
                 }
 #endif
 #if defined(FEATURE_SIMD) && defined(TARGET_XARCH)
                 else if (indir->TypeIs(TYP_SIMD16, TYP_SIMD32) && (genTypeSize(indir) * 2 == genTypeSize(varDsc)) &&
-                         ((offset % genTypeSize(indir)) == 0))
+                         ((offset % genTypeSize(indir)) == 0) && m_compiler->IsBaselineSimdIsaSupported())
```
[the remark this answers is not in the document]
Comment (EgorBo) [reply src/coreclr/jit/lclmorph.cpp:1200]: Ah, I see, you mean Baseline might not be enough for SIMD32 ?

[the remark this answers is not in the document]
Comment (jakobbotsch) [reply src/coreclr/jit/lclmorph.cpp:1200]: Shouldn't the logic instead be:
`((indir->TypeIs(TYP_SIMD16) && m_compiler->IsBaselineSimdIsaSupported()) || (indir->TypeIs(TYP_SIMD32) && m_compiler->compOpportunisticallyDependsOn(InstructionSet_AVX))` (or something like that)?

[the remark this answers is not in the document]
Comment (jakobbotsch) [reply src/coreclr/jit/lclmorph.cpp:1200]: > Ah, I see, you mean Baseline might not be enough for SIMD32 ?

Right.

[the remark this answers is not in the document]
Comment (jakobbotsch) [reply src/coreclr/jit/lclmorph.cpp:1200]: It should probably actually be:
```
((indir->TypeIs(TYP_SIMD16) && m_compiler->compOpportunisticallyDependsOn(InstructionSet_AVX)) || (indir->TypeIs(TYP_SIMD32) && m_compiler->IsBaselineVector512IsaSupportedOpportunistically())
```

I'm a bit confused why we can have SIMD32 typed locals when SIMD32 is not being treated as a primitive, but that's a separate discussion.

File: src/coreclr/jit/lclmorph.cpp [hunk 2]
```diff
@@ -1166,41 +1166,76 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
...
+#if defined(TARGET_XARCH)
+                else if (indir->TypeIs(TYP_SIMD32))
+                {
+                    simdElementAccessAllowed = m_compiler->compOpportunisticallyDependsOn(InstructionSet_AVX);
+                }
+                else if (indir->TypeIs(TYP_SIMD64))
+                {
+                    simdElementAccessAllowed = m_compiler->compOpportunisticallyDependsOn(InstructionSet_AVX512F);
                 }
+#endif
```
Comment (jakobbotsch) [inline src/coreclr/jit/lclmorph.cpp:1189]: This doesn't look right. `indir->TypeIs(TYP_SIMD32)` means that we are going to use AVX512 intrinsics to extract/insert a half into an AVX512 register, so this case should check for AVX512. Similar for `indir->TypeIs(TYP_SIMD16)`.

Why does it need to be so complicated? Can `indir->TypeIs(TYP_SIMD16, TYP_SIMD32)` not just be changed to `(indir->TypeIs(TYP_SIMD16) && m_compiler->compOpportunisticallyDependsOn(InstructionSet_AVX)) || (indir->TypeIs(TYP_SIMD32) && m_compiler->compOpportunisticallyDependsOn(InstructionSet_AVX512F))`?

Comment (EgorBo) [reply src/coreclr/jit/lclmorph.cpp:1189]: @jakobbotsch ah, I misunderstood the logic here a bit, gotcha

Comment (jakobbotsch) [reply src/coreclr/jit/lclmorph.cpp:1189]: No worries -- I wonder if the logic could even just be `genTypeSize(varDsc) <= getMaxVectorByteLength()`

Comment (EgorBo) [reply src/coreclr/jit/lclmorph.cpp]: Done!

**Review by jakobbotsch — APPROVED** [review]

**Pull request merged**